src/components/ConnectionDialog.tsx

```diff
@@ -9,19 +9,22 @@ export type Props = {
     onSubmit: (connection: Connection) => void,
 }
 
-export default function ConnectionDialog({isOpen, onClose, onSubmit}: Props) {    
+export default function ConnectionDialog({isOpen, onClose, onSubmit}: Props) {
+    const [connectionName, setConnectionName] = useState("")
     const [baseUrl, setBaseUrl] = useState("")
     const [auth, setAuth] = useState("")
 
     const handleOpening = () => {
+        setConnectionName("")
         setBaseUrl("")
         setAuth("")
     }
     const handleSubmit = () => {
         if (isValid()) {
+            const name = connectionName.trim()
```

Comment:
Maybe unnecessary but this is to try and avoid names that are all spaces. Alternative would be to change the `onChange` handler to `e => setConnectionName(e.target.value.trimStart())` which should achieve the same effect while being a bit more honest to the user in the dialog box form.

Reply:
I think I prefer to clean it up before saving, rather than manipulating the form data while the user is typing.